### ISSUE TITLE
java.util.MissingResourceException on HikariCP + MS SQL Server

### ISSUE THREAD

**Issue by dnaumovich** [brettwooldridge/HikariCP#1428]
#### Environment
```
HikariCP version: 3.3.1
JDK version     : 1.8.0_162
Database        : MS SQL Server
Driver version  : 6.2.2
```
Hello. I have a Java Servlet-based application which uses HikariCP for obtaining connections to MS SQL Server database. Everything works, connection arrives, JDBC calls are performed, but in the tomcat logs there's such an exception which is thrown every 7-8 seconds (approximately).

```
[HikariPool-27 connection adder] ERROR com.zaxxer.hikari.pool.HikariPool - HikariPool-27 - Error thrown while acquiring connection from data source
java.util.MissingResourceException: Can't find bundle for base name com.microsoft.sqlserver.jdbc.SQLServerResource, locale en_US
	at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:1573)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1396)
	at java.util.ResourceBundle.getBundle(ResourceBundle.java:782)
	at com.microsoft.sqlserver.jdbc.SQLServerResource.getResource(SQLServerResource.java:19)
	at com.microsoft.sqlserver.jdbc.SQLServerException.getErrString(SQLServerException.java:121)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.setSchema(SQLServerConnection.java:4836)
	at com.zaxxer.hikari.pool.PoolBase.setupConnection(PoolBase.java:413)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:355)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:201)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:473)
	at com.zaxxer.hikari.pool.HikariPool.access$100(HikariPool.java:71)
	at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:727)
	at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:713)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
java.util.MissingResourceException: Can't find bundle for base name com.microsoft.sqlserver.jdbc.SQLServerResource, locale en_US
	at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:1573)
	at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1396)
	at java.util.ResourceBundle.getBundle(ResourceBundle.java:782)
	at com.microsoft.sqlserver.jdbc.SQLServerResource.getResource(SQLServerResource.java:19)
	at com.microsoft.sqlserver.jdbc.SQLServerException.getErrString(SQLServerException.java:121)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.setSchema(SQLServerConnection.java:4836)
	at com.zaxxer.hikari.pool.PoolBase.setupConnection(PoolBase.java:413)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:355)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:201)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:473)
	at com.zaxxer.hikari.pool.HikariPool.access$100(HikariPool.java:71)
	at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:727)
	at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:713)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

I couldn't find anything related to that over the web.
Here's my HikariCP configuration:
```
private static HikariConfig config = new HikariConfig();
    private static HikariDataSource ds;

    static {
        String address = Config.getInstance().getSettingString(ServerSetting.DB_ADDRESS);
        String dbName = Config.getInstance().getSettingString(ServerSetting.DB_NAME);

        String jdbcUrl = format("jdbc:sqlserver://%s;databaseName=%s;SelectMethod=Cursor;integratedSecurity=false;cacheMetaData=true", address, dbName);
        config.setJdbcUrl(jdbcUrl);
        config.setDriverClassName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
        config.setUsername(Config.getInstance().getSettingString(ServerSetting.DB_USERNAME));
        config.setPassword(Config.getInstance().getSettingString(ServerSetting.DB_PASSWORD));
        ds = new HikariDataSource(config);
    }
```

I wouldn't like to create a bundle properties files for that but this exception messes tomcat logs and it's hard to follow the requires logs.
Thank you.

**Comment by ljluestc** [comment]
```

import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;

public class DataSourceProvider {

    private static HikariConfig config = new HikariConfig();
    private static HikariDataSource ds;

    static {
        String address = Config.getInstance().getSettingString(ServerSetting.DB_ADDRESS);
        String dbName = Config.getInstance().getSettingString(ServerSetting.DB_NAME);

        String jdbcUrl = String.format("jdbc:sqlserver://%s;databaseName=%s;SelectMethod=Cursor;integratedSecurity=false;cacheMetaData=true", address, dbName);
        config.setJdbcUrl(jdbcUrl);
        config.setDriverClassName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
        config.setUsername(Config.getInstance().getSettingString(ServerSetting.DB_USERNAME));
        config.setPassword(Config.getInstance().getSettingString(ServerSetting.DB_PASSWORD));
        
        // Optional: Configure additional settings as needed
        config.setMaximumPoolSize(10); // example setting
        config.setMinimumIdle(5); // example setting
        
        ds = new HikariDataSource(config);
    }

    public static HikariDataSource getDataSource() {
        return ds;
    }
}
```
